### PR TITLE
test(qa): use admin endpoint to verify snapshots 

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

Since this is a qa test, it is better to use the existing apis rather than exposing the internal implementations details such as snapshot directory in the tests. This PR is in preparation to #6666, making tests less dependent on the internals so that it will be easier to refactor. 

It was not possible to use http client for admin endpoint because the brokers are not started as Spring boot applications. Hence the endpoints are not available. Instead the test directly used `BrokerAdminService` to query the status of the partitions.

## Related issues

#6666 

Related #snapshot

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/SnapshotReplicationTest.java [hunk 1]
```diff
@@ -128,21 +120,13 @@ public void shouldReplicateSnapshotsOnLeaderChange() {
...
-      final var checksums = createSnapshotDirectoryChecksums(broker);
-      brokerSnapshotChecksums.put(broker.getConfig().getCluster().getNodeId(), checksums);
-    }
-    return brokerSnapshotChecksums;
+    final var snapshotAtOldLeader = clusteringRule.getSnapshot(oldLeaderId).orElseThrow();
+    assertThat(otherBrokers)
+        .allSatisfy(
+            broker ->
+                assertThat(clusteringRule.getSnapshot(broker).orElseThrow())
+                    .as("All brokers have same snapshot")
```
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/SnapshotReplicationTest.java:128]: I like it :+1:

**Review by npepinpe — APPROVED** [review]
bors r+

Thanks!

**Comment by npepinpe** [comment]
bors retry

**Comment by deepthidevaki** [comment]
@npepinpe One test was failing because using string snapshot ids to compare does not work. I have reverted those changes and use `SnapshotId` generated by parsing the string using `FileBasedSnapshotMetadata`.

**Comment by npepinpe** [comment]
Can you elaborate why using the string IDs doesn't work? Shouldn't they be the same?

**Comment by deepthidevaki** [comment]
> Can you elaborate why using the string IDs doesn't work? Shouldn't they be the same?

We have to compare if one snapshot is greater than the other. With string comparison is not the same as snapshot id comparisons. String comparisons only work for equality.

**Pull request merged**